.github/workflows/tests.yaml

```diff
@@ -0,0 +1,85 @@
+name: Test Suite
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, windows-latest]
+        php: [ "7.0", "7.1", "7.2", "7.3", "7.4" ]
+        phpunit-suffix: [phpunit]
+        include:
+          - platform: ubuntu-latest
+            php: "5.5"
+            phpunit-suffix: phpunit-php5
+          - platform: ubuntu-latest
+            php: "5.6"
+            phpunit-suffix: phpunit-php5
+          - platform: windows-latest
+            php: "5.5"
+            phpunit-suffix: phpunit-php5
+          - platform: windows-latest
+            php: "5.6"
+            phpunit-suffix: phpunit-php5
+    name: PHP ${{ matrix.php }} Unit Test (${{ matrix.platform }})
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php }}
+        tools: pecl
+        extensions: grpc
+    - name: Install Dependencies
+      uses: nick-invision/retry@v2
+      with:
+        timeout_minutes: 10
+        max_attempts: 3
+        command: composer --no-interaction --no-ansi --no-progress update
+    - name: Run Unit Test Suite
+      run: |
+        vendor/bin/phpunit -c ${{ matrix.phpunit-suffix }}.xml.dist
+    - if: ${{ matrix.platform != 'windows-latest' }}
```

Comment:
We don't run snippets tests on windows because an [error is being thrown in PHPdocumentor](https://github.com/googleapis/google-cloud-php/runs/6565342495?check_suite_focus=true#step:6:9):

```
PHP Fatal error:  Uncaught RuntimeException: ?array is not a collection in D:\a\google-cloud-php\google-cloud-php\vendor\phpdocumentor\type-resolver\src\TypeResolver.php:613
```